### PR TITLE
#184581358 expired product stop listing

### PR DESCRIPTION
**Pull request by blackd44**
#### What does this PR do?
make the expired product not available on the market

#### Description of Task to be completed?
the expired product are removed on the available products
before being removed the seller is given the email in a give amount of time to be warned
and on the exact day of expiration the product is removed and then give the seller the email again

#### How should this be manually tested?
after cloning the repo,
- in terminal add `npm install`
- add `.env` in your repo and assign the variables as there are in `.env.example` file
- run in termainal `npm run migrate:all` to setup the database
- run in termainal `npm start` to start the project
- in your browser add `http://localhost:<your port>/api-docs` for documentantion

#### Any background context you want to provide?
- Crone jobs
- node schedule
- node mailer

#### What are the relevant pivotal tracker/Trello stories?
https://www.pivotaltracker.com/n/projects/2632830/stories/184581358

#### Screenshots (if appropriate)
Screen Shot 2023-03-21 at 17 08 11
Screen Shot 2023-03-21 at 17 12 18

#### Questions:
n/a

### CODE REVIEW

**Review by edwinnambaje — APPROVED** [review]
LGTM

File: src/helpers/expiredProduct.js [hunk 1]
```diff
@@ -0,0 +1,75 @@
...
+          email: user.email,
+          subject: `Expiry Date of ${product.name} is Approaching`,
+          content: html
+        })
+      );
+    } else {
+      scheduleExpirationEmail(new Date(product.exp_date), () => checkExpiredProduct(product));
+    }
+  } catch (error) {
+    // nothing
```
Comment (nshutijonathan) [inline src/helpers/expiredProduct.js]: you can `return error.message` just in case

File: src/middlewares/productExpAll.js [hunk 2]
```diff
@@ -0,0 +1,15 @@
...
+import checkExpiredProduct from '../helpers/expiredProduct';
+
+export default async function beforeCheckExpiredProduct() {
+  try {
+    const allProducts = await products.findAll();
+    allProducts.forEach(async (product) => {
+      checkExpiredProduct(product);
+    });
+  } catch (error) {
+    console.log(error);
```
Comment (nshutijonathan) [inline src/middlewares/productExpAll.js]: `return error.message` ,remember console is for debugging only

**Review by nshutijonathan — CHANGES_REQUESTED** [review]
I left some comments

**Review by Patienceineza — APPROVED** [review]
LGTM

**Review by INValentin — APPROVED** [review]
LGTM

**Review by Zeden12 — APPROVED** [review]
LGTM

**Review by nshutijonathan — APPROVED** [review]
LGTM

**Review by UmuhireJessie — APPROVED** [review]
Good job!

**Pull request merged**